src/app/content/components/utils/attachHighlight.ts

```diff
@@ -1,16 +1,22 @@
 import Highlighter, { Highlight, SerializedHighlight } from '@openstax/highlighter';
 import Sentry from '../../../../helpers/Sentry';
 
-const attachHighlight = (
-  highlight: Highlight | SerializedHighlight,
-  highlighter: Highlighter
+const attachHighlight = <T extends Highlight | SerializedHighlight>(
+  highlight: T,
+  highlighter: Highlighter,
+  errorMsg?: (highlight: T) => string
 ) => {
-  highlighter.highlight(highlight);
-  const result = highlighter.getHighlight(highlight.id);
-  if (!result || !result.isAttached()) {
-    Sentry.captureException(new Error(`Highlight with id: ${highlight.id} has not been attached.`));
+  try {
+    highlighter.highlight(highlight);
+    const result = highlighter.getHighlight(highlight.id);
+    if (!result || !result.isAttached()) {
+      throw new Error(`Highlight with id: ${highlight.id} has not been attached.`);
+    }
+    return result;
+  } catch (e) {
+    const error = errorMsg ? new Error(errorMsg(highlight)) : e;
+    Sentry.captureException(error);
```

Comment:
this logic seems very odd, first, are there any exceptions you're trying to catch here other than the one thrown above? why not just capture it there instead of throwing it? also, why default the message above and then override it here, why not put the custom message in on line 13?

Reply:
I'm pretty sure that `higlighter.highlight` throws error like `cannot read property matches of null` when search results fails to highlight too. If i were to default it there then it is possible that it wouldn't actually record that error but rather the one i just mentioned.
cc. @TomWoodward

Reply:
if `highlight` throws `cannot read property matches of null` that is an indication of another issue that we'd like to have in sentry, and your change that you have in place will replace it with this more generic error.